src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -73,72 +71,87 @@ internal partial class CodeFixService : ICodeFixService
             _configurationProvidersMap = GetConfigurationProvidersPerLanguageMap(configurationProviders);
         }
 
-        public async Task<FirstDiagnosticResult> GetMostSevereFixableDiagnosticAsync(
-            Document document, TextSpan range, CancellationToken cancellationToken)
+        private Func<string, bool>? GetShouldIncludeDiagnosticPredicate(
+            Document document,
+            CodeActionRequestPriority priority)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            // For 'CodeActionPriorityRequest.Normal' or 'CodeActionPriorityRequest.Low', we do not compute
+            // suppression/configuration fixes, those fixes have a dedicated request priority
+            // 'CodeActionPriorityRequest.Lowest'.
+
+            // Hence, for Normal or Low priority, we only need to execute analyzers which can report at least one
+            // fixable diagnostic that can have a non-suppression/configuration fix. Note that for
+            // 'CodeActionPriorityRequest.High', we only run compiler analyzer, which always has fixable
+            // diagnostics, so we can pass in null. 
+
+            if (!(priority is CodeActionRequestPriority.Normal or CodeActionRequestPriority.Low))
+                return null;
+
+            var hasWorkspaceFixers = TryGetWorkspaceFixersMap(document, out var workspaceFixersMap);
+            var projectFixersMap = GetProjectFixers(document.Project);
 
-            if (document == null || !document.IsOpen())
+            return id =>
             {
-                return default;
-            }
+                if (hasWorkspaceFixers && workspaceFixersMap!.Value.ContainsKey(id))
+                    return true;
 
-            using var _ = ArrayBuilder<DiagnosticData>.GetInstance(out var diagnostics);
-            using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                return projectFixersMap.ContainsKey(id);
+            };
+        }
+
+        public async Task<FirstFixResult> GetMostSevereFixAsync(
+            Document document, TextSpan range, CodeActionRequestPriority priority, CodeActionOptions options, CancellationToken cancellationToken)
+        {
+            var (allDiagnostics, upToDate) = await _diagnosticService.TryGetDiagnosticsForSpanAsync(
+                document, range, GetShouldIncludeDiagnosticPredicate(document, priority),
+                includeSuppressedDiagnostics: false, priority, cancellationToken).ConfigureAwait(false);
 
+            var spanToDiagnostics = ConvertToMap(allDiagnostics);
```

Comment:
the start of this is extremely similar to StreamFixesAsync.  Both start similarly (though one calls TryGetDaignostics, and one called GetDiagnostics), but how they process hte remainder is different.